Last-Update: 2013-12-28
Forwarded: not-needed
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
From: Roland Stigge <stigge@antcom.de>
Bug-Debian: http://bugs.debian.org/701645
Description: fix FTBFS on powerpc, ppc64 and powerpcspe.

--- a/src/framework/tcore/kcpuinfo.cpp
+++ b/src/framework/tcore/kcpuinfo.cpp
@@ -32,8 +32,11 @@
  *   You should have received a copy of the GNU General Public License     *
  *   along with this program.  If not, see <http://www.gnu.org/licenses/>. *
  ***************************************************************************/
 
+#include <signal.h>
+#include <setjmp.h>
+
 #include "kcpuinfo.h"
 
 #if defined(__GNUC__) || defined(__INTEL_COMPILER)
 #  define HAVE_GNU_INLINE_ASM
