Last-Update: 2013-05-13
Forwarded: no
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Description: always build "tupi.desktop" (not just on linux).

--- a/qonf/configure.rb
+++ b/qonf/configure.rb
@@ -221,9 +221,8 @@
       launcher = File.open("launcher/tupi", "w") { |f|
         f << newfile
       }
 
-      if RUBY_PLATFORM.downcase.include?("linux")
         newfile = "[Desktop Entry]\n"
         # newfile += "Encoding=UTF-8\n"
         newfile += "Name=Tupi: 2D Magic\n"
         newfile += "Name[es]=Tupí: Magia 2D\n"
@@ -242,9 +241,8 @@
 
         File.open("launcher/tupi.desktop", "w") { |f|
           f << newfile
         }
-      end
 
       newmakefile = ""
       File.open("src/components/help/help/css/tupi.ini", "r") { |f|
         lines = f.readlines
