Last-Update: 2012-02-03
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
From: Gustav Gonzalez <info@maefloresta.com>
Origin: upstream, backported
Forwarded: not-needed
Description: fix hardcoded path to binary in crash handler


--- a/src/framework/tcore/tapplicationproperties.cpp
+++ b/src/framework/tcore/tapplicationproperties.cpp
@@ -44,6 +44,7 @@
 struct TApplicationProperties::Private
 {
     QString homeDir;
+    QString binDir;
     QString shareDir;
     QString dataDir;
     QString themeDir;
@@ -64,6 +65,16 @@
     delete k;
 }
 
+void TApplicationProperties::setHomeDir(const QString &path)
+{
+    k->homeDir = path;
+}
+
+void TApplicationProperties::setBinDir(const QString &path)
+{
+    k->binDir = path;
+}
+
 void TApplicationProperties::setShareDir(const QString &path)
 {
     k->shareDir = path;
@@ -74,11 +85,6 @@
     k->dataDir = path;
 }
 
-void TApplicationProperties::setHomeDir(const QString &path)
-{
-    k->homeDir = path;
-}
-
 void TApplicationProperties::setThemeDir(const QString &path)
 {
     k->themeDir = path;
@@ -114,6 +120,16 @@
     k->revision = path;
 }
 
+QString TApplicationProperties::homeDir() const
+{
+    return k->homeDir + QDir::separator();
+}
+
+QString TApplicationProperties::binDir() const
+{
+    return k->binDir + QDir::separator();
+}
+
 QString TApplicationProperties::shareDir() const
 {
     if (k->shareDir.isEmpty())
@@ -135,11 +151,6 @@
     return k->dataDir;
 }
 
-QString TApplicationProperties::homeDir() const
-{
-    return k->homeDir + QDir::separator();
-}
-
 QString TApplicationProperties::themeDir() const
 {
     if (k->themeDir.isEmpty())
--- a/src/framework/tcore/tapplicationproperties.h
+++ b/src/framework/tcore/tapplicationproperties.h
@@ -50,9 +50,10 @@
         virtual ~TApplicationProperties();
 
     public:
+        void setHomeDir(const QString &path);
+        void setBinDir(const QString &path);
         void setShareDir(const QString &path);
         void setDataDir(const QString &path);
-        void setHomeDir(const QString &path);
         void setThemeDir(const QString &path);
         void setPluginDir(const QString &path);
         void setCacheDir(const QString &path);
@@ -61,9 +62,10 @@
         void setCodeName(const QString &path);
         void setRevision(const QString &path);
 
+        virtual QString homeDir() const;
+        virtual QString binDir() const;
         virtual QString shareDir() const;
         virtual QString dataDir() const;
-        virtual QString homeDir() const;
         virtual QString themeDir() const;
         virtual QString pluginDir() const;
         virtual QString configDir() const;
--- a/src/framework/tcore/tglobal.h
+++ b/src/framework/tcore/tglobal.h
@@ -48,10 +48,11 @@
 
 // #include "tapplicationproperties.h"
 
+#define HOME_DIR kAppProp->homeDir()
+#define BIN_DIR kAppProp->binDir()
 #define SHARE_DIR kAppProp->shareDir()
 #define DATA_DIR kAppProp->dataDir()
 #define THEME_DIR kAppProp->themeDir()
-#define HOME_DIR kAppProp->homeDir()
 #define CONFIG_DIR kAppProp->configDir()
 #define PLUGINS_DIR kAppProp->pluginDir()
 #define CACHE_DIR kAppProp->cacheDir()
--- a/src/shell/ktcrashhandler.cpp
+++ b/src/shell/ktcrashhandler.cpp
@@ -309,7 +309,7 @@
 #ifdef K_DEBUG
         if (QFile::exists("/usr/bin/sudo") && QFile::exists("/usr/bin/gdb")) {
             QString gdb;
-            gdb = "/usr/bin/sudo /usr/bin/gdb -n -nw -batch -ex where " + HOME_DIR + "bin/tupi.bin --pid=";
+            gdb = "/usr/bin/sudo /usr/bin/gdb -n -nw -batch -ex where " + BIN_DIR + "tupi.bin --pid=";
             gdb += QString::number(::getppid());
             bt = runCommand(gdb);
 
@@ -318,7 +318,7 @@
             bt = bt.simplified();
         } 
 #endif
-        execInfo = runCommand("file " + HOME_DIR + "bin/tupi.bin");
+        execInfo = runCommand("file " + BIN_DIR + "tupi.bin");
 
         // Widget
         QDesktopWidget desktop;
--- a/src/shell/main.cpp
+++ b/src/shell/main.cpp
@@ -129,6 +129,7 @@
     }
 
     kAppProp->setHomeDir(TCONFIG->value("Home").toString());
+    kAppProp->setBinDir(QString::fromLocal8Bit(::getenv("TUPI_BIN")));
     kAppProp->setPluginDir(QString::fromLocal8Bit(::getenv("TUPI_PLUGIN")));
     kAppProp->setShareDir(QString::fromLocal8Bit(::getenv("TUPI_SHARE")));
 
