--- tupi-0.1+git12.orig/src/framework/tcore/kcpuinfo.cpp
+++ tupi-0.1+git12/src/framework/tcore/kcpuinfo.cpp
@@ -33,6 +33,9 @@
  *   along with this program.  If not, see <http://www.gnu.org/licenses/>. *
  ***************************************************************************/
 
+#include <signal.h>
+#include <setjmp.h>
+
 #include "kcpuinfo.h"
 
 #if defined(__GNUC__) || defined(__INTEL_COMPILER)
