Last-Update: 2016-02-13
Forwarded: yes
Bug-Debian: https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=810609
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Description: correct library name

--- a/qonf/test.rb
+++ b/qonf/test.rb
@@ -102,9 +102,9 @@
                          extraInclude = quazipDir + "/include"
                          qmakeLine = "'LIBS += #{extraLib}'"
                          qmakeLine += " 'INCLUDEPATH += #{extraInclude}'"
                       else
-                         extraLib = "-lquazip-qt5"
+                         extraLib = "-lquazip5"
                          qmakeLine = "'LIBS += #{extraLib}'"
                       end
                    else
                       if File.dirname(@rules).end_with?("theora")
@@ -175,9 +175,9 @@
 
         if conf.hasArgument?("with-quazip")
            config.addLib("-lquazip")
         else
-           config.addLib("-lquazip-qt5")
+           config.addLib("-lquazip5")
         end
 
         # config.addLib("-lquazip-qt5")
         
