Last-Update: 2015-01-02
Forwarded: yes
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Description: drop stupid quazip detection.

--- a/qonf/test.rb
+++ b/qonf/test.rb
@@ -102,13 +102,9 @@
                          extraInclude = quazipDir + "/include"
                          qmakeLine = "'LIBS += #{extraLib}'"
                          qmakeLine += " 'INCLUDEPATH += #{extraInclude}'"
                       else
-                         if parser.os.eql? "14.10" 
-                            extraLib = "-lquazip-qt5"
-                         else
-                            extraLib = "-lquazip"
-                         end
+                         extraLib = "-lquazip-qt5"
                          qmakeLine = "'LIBS += #{extraLib}'"
                       end
                    else
                       if File.dirname(@rules).end_with?("theora")
@@ -176,13 +172,9 @@
         parser.libs.each { |lib|
             config.addLib(lib)
         }
 
-        if parser.os.eql? "14.10"
-           config.addLib("-lquazip-qt5")
-        else
-           config.addLib("-lquazip")
-        end
+        config.addLib("-lquazip-qt5")
         
         parser.defines.each { |define|
             config.addDefine(define)
         }
