Last-Update: 2015-01-04
Forwarded: no
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Description: enable verbose build.

--- a/configure.rb
+++ b/configure.rb
@@ -158,9 +158,8 @@
        config.addDefine("K_DEBUG")
        print "[ \033[92mON\033[0m ]\n"
     else
        config.addDefine("K_NODEBUG")
-       config.addOption("silent")
        print "[ \033[91mOFF\033[0m ]\n"
     end
 
     if File.exists?('/etc/canaima_version')
